butterfly/QueryLayer/TileQuery.py

```diff
@@ -82,12 +97,32 @@ def source_bounds(self):
     def preload_source(self):
         cache_meta = self.RUNTIME.CACHE.META
         # Preload the metadata from the source
-        keywords = self.source_class.preload_source(self)
+        keywords = self.valid_source
+
         # Get the size of this dicitonary for the cache
-        keywords[cache_meta.NAME] = np.uint32(sys.getsizeof({}))
+        dict_size = np.uint32(sys.getsizeof({}))
+        keywords[cache_meta.NAME] = dict_size
         # calculate the size
         for key in keywords.keys():
             n_bytes = sys.getsizeof(keywords[key])
             keywords[cache_meta.NAME] += n_bytes
         # Return keywords for cache and dataQuery
         return keywords
+
+    @property
+    def valid_source(self):
+        my_source = self.RUNTIME.IMAGE.SOURCE
+        # Validate the source of self.path
+        for name in self.source_list:
+            source = self.get_source(name)
+            # Ask if source can load self path
+            keywords = source.preload_source(self)
+            if keywords:
+                # Set valid source
+                keywords[my_source.NAME] = name
+                return keywords
+        # return empty source keyword
+        return {my_source.NAME: None}
+
```

Comment:
Here we set the source to `None`. This should be used for error handling later in `Query.update_source`